metrics/internal/collectors/object-bucket.go

```diff
@@ -0,0 +1,167 @@
+package collectors
+
+import (
+	"context"
+	"fmt"
+
+	rgwadmin "github.com/ceph/go-ceph/rgw/admin"
+	libbucket "github.com/kube-object-storage/lib-bucket-provisioner/pkg/apis/objectbucket.io/v1alpha1"
+	bktclient "github.com/kube-object-storage/lib-bucket-provisioner/pkg/client/clientset/versioned"
+	"github.com/openshift/ocs-operator/metrics/internal/options"
+	"github.com/prometheus/client_golang/prometheus"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
+	cephv1listers "github.com/rook/rook/pkg/client/listers/ceph.rook.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+const (
+	bucketProvisionerName = "ceph.rook.io-bucket"
+	rgwServiceName        = "rook-ceph-rgw"
+	svcDNSSuffix          = "svc"
+	prometheusUserName    = "prometheus-user"
+	endPoint              = "Endpoint"
+	accessKey             = "AccessKey"
+	secretKey             = "SecretKey"
+	cephUser              = "cephUser"
+)
+
+var _ prometheus.Collector = &OBCollector{}
+
+// OBCollector is a custom collector for CephObjectStore Custom Resource
+type OBCollector struct {
+	OBSizeConsumption *prometheus.Desc
+	OBObjectCount     *prometheus.Desc
+	Informer          cache.SharedIndexInformer
+	AllowedNamespaces []string
+	bktclient         bktclient.Interface
+	rookclient        rookclient.Interface
+	k8sclient         kubernetes.Interface
+}
+
+// NewOBCollector constructs a collector
+func NewOBCollector(opts *options.Options) *OBCollector {
+
+	sharedIndexInformer := CephObjectStoreInformer(opts)
+
+	return &OBCollector{
+		OBSizeConsumption: prometheus.NewDesc(
```

Comment:
I see this PR is exposing only 2 metrics, 1 for the max size of the quota and the other for the max number of objects. 

AFAIU, the objective of the metrics is to create alerts that inform users when the obc crosses a threshold.

Thus there is a need for the following metrics: 

Quota in bytes: 

- Used : obc_used_bytes
- Max: obc_max_bytes

Quota in number of objects: 

- Used : obc_objects_total
- Max : obc_objects_max

Reply:
For adding other two metrics I am waiting on this https://github.com/ceph/go-ceph/pull/546 in go-ceph